src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/updateRouterAttributesFromUserStoreContentLocale.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import pathToRegexp from 'path-to-regexp';
+import {toJS} from 'mobx';
+import type {AttributeMap, Route, UpdateAttributesHook} from '../../services/Router/types';
+import userStore from './userStore';
+
+const updateRouterAttributesFromUserStoreContentLocale: UpdateAttributesHook = function(
+    route: Route,
+    attributes: AttributeMap
+) {
+    // do nothing when locale is explicit set
+    if (attributes.locale) {
+        return attributes;
+    }
+
+    const keys = [];
+    pathToRegexp(route.path, keys);
+    const keyNames = keys.map((key) => key.name);
+
+    // do nothing when the route does not require a locale
+    if (!keyNames.includes('locale')) {
+        return attributes;
+    }
+
+    const locales = toJS(route.options.locales);
+
+    // set content locale if route accept the current content locale
+    // or does not specify which locales it accepts e.g.: PageList
```

Comment:
Then the expected behaviour does not longer work. We need todo this here else you will land again on the default webspace locale everytime.

Reply:
I don't think this is the case if you add setting the locale to the `getDerivedRouteAttributes` method of the `PageList` instead of doing it here... But I know that you won't like that. I'll just keep that comment open, and test this part extensively at the end, to see if something is not working as expected.